arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,903 @@
+                        "type": "string"
+                    }
+                }
+            }
+        },
+        "CommitmentAssociation": {
+            "type": "object",
+            "description": "Represents the association between a commitment plan and some other resource, such as a Machine Learning web service.",
+            "properties": {
+                "etag": {
+                    "type": "string",
+                    "description": "An entity tag used to enforce optimistic concurrency."
+                },
+                "id": {
+                    "type": "string",
+                    "description": "The commitment association ARM resource ID."
+                },
+                "location": {
+                    "type": "string",
+                    "description": "The resource location."
+                },
+                "name": {
+                    "type": "string",
+                    "description": "The resource name.",
+                    "readOnly": true
+                },
+                "properties": {
+                    "$ref": "#/definitions/CommitmentAssociationProperties"
+                },
+                "type": {
+                    "type": "string",
+                    "description": "The type of the resource: Microsoft.MachineLearning/commitmentPlans/commitmentAssociations",
+                    "readOnly": true
+                }
+            }
+        },
+        "CommitmentAssociationProperties": {
+            "type": "object",
+            "description": "Properties of a commitment association resource.",
+            "properties": {
+                "associatedResourceId": {
+                    "type": "string",
+                    "description": "The ID of the resource this association points to, such as the ARM ID of a Machine Learning web service."
+                },
+                "commitmentPlanId": {
+                    "type": "string",
+                    "description": "The ARM ID of the parent commitment plan resource."
+                },
+                "creationDate": {
+                    "format": "date-time",
+                    "type": "string",
+                    "description": "The date at which this commitment association was created."
+                }
+            }
+        },
+        "ResourceSku": {
+            "type": "object",
+            "description": "The SKU of a resource.",
+            "properties": {
+                "capacity": {
+                    "format": "int32",
+                    "type": "integer",
+                    "description": "The scale-out capacity of the resource. 1 is 1x, 2 is 2x, etc. This impacts the quantities and cost of any commitment plan resource."
+                },
+                "name": {
+                    "type": "string",
+                    "description": "The SKU name. Along with tier, uniquely identifies the SKU."
+                },
+                "tier": {
```

Comment:
classic case to be an enum. How is the user supposed to know what should be provided over here